### PR TITLE
Implement set-security-context feature for affinity assistant containers

### PR DESCRIPTION
**Pull request by kristofferchr**
Solves https://github.com/tektoncd/pipeline/issues/8181

Needed for this issue: https://github.com/tektoncd/pipeline/issues/8183

# Changes

Added container securityContext for affinity assistant when feature flag `set-security-context` is set to true. 
Implemented for both windows and linux OS. )

# Submitter Checklist

As the author of this PR, please check off the items in this checklist:

- [x] Has [Docs](https://github.com/tektoncd/community/blob/main/standards.md#docs) if any changes are user facing, including updates to minimum requirements e.g. Kubernetes version bumps
- [x] Has [Tests](https://github.com/tektoncd/community/blob/main/standards.md#tests) included if any functionality added or changed
- [x] [pre-commit](https://github.com/tektoncd/pipeline/blob/main/DEVELOPMENT.md#install-tools) Passed
- [x] Follows the [commit message standard](https://github.com/tektoncd/community/blob/main/standards.md#commits)
- [x] Meets the [Tekton contributor standards](https://github.com/tektoncd/community/blob/main/standards.md) (including functionality, content, code)
- [x] Has a kind label. You can add one by adding a comment on this PR that contains `/kind <type>`. Valid types are bug, cleanup, design, documentation, feature, flake, misc, question, tep
- [x] Release notes block below has been updated with any user facing changes (API changes, bug fixes, changes requiring upgrade notices or deprecation warnings). See some examples of [good release notes](https://github.com/tektoncd/community/blob/main/standards.md#good-release-notes).
- [x] Release notes contains the string "action required" if the change requires additional action from users switching to the new release

# Release Notes

```release-note
Affinity Assistant containers will now have a securityContext when feature flag `set-security-context` is enabled in ConfigMap `feature-flags`.
```

### CODE REVIEW

File: pkg/reconciler/pipelinerun/affinity_assistant.go [hunk 1]
```diff
@@ -23,6 +23,8 @@ import (
 	"errors"
 	"fmt"
 
+	pipelinePod "github.com/tektoncd/pipeline/pkg/pod"
+
```
Comment (vdemeester) [inline pkg/reconciler/pipelinerun/affinity_assistant.go:27]: nit: extra space.

File: pkg/reconciler/pipelinerun/affinity_assistant_test.go [hunk 2]
```diff
@@ -22,6 +22,8 @@ import (
 	"fmt"
 	"testing"
 
+	pipelinePod "github.com/tektoncd/pipeline/pkg/pod"
+
```
Comment (vdemeester) [inline pkg/reconciler/pipelinerun/affinity_assistant_test.go:26]: nit: same here

**Review by vdemeester — APPROVED** [review]

File: pkg/reconciler/pipelinerun/affinity_assistant.go (hunk 1 above)
Comment (kristofferchr) [reply pkg/reconciler/pipelinerun/affinity_assistant.go]: Good catch, fixed

File: pkg/reconciler/pipelinerun/affinity_assistant_test.go (hunk 2 above)
Comment (kristofferchr) [reply pkg/reconciler/pipelinerun/affinity_assistant_test.go]: fixed

**Comment by kristofferchr** [comment]
@vdemeester is this /ok-to-test ?

**Comment by kristofferchr** [comment]
@chitrangpatel this needs a new lgtm, had to rebase.

**Review by vdemeester — APPROVED** [review]
/lgtm

**Comment by kristofferchr** [comment]
/test pull-tekton-pipeline-integration-tests
--


<br class="Apple-interchange-newline">

**Comment by kristofferchr** [comment]
@vdemeester need another /lgtm rebase changes again.

**Pull request merged**